### PR TITLE
[Diffusion] Add TeaCache support for HiDream-I1-Full

### PR DESCRIPTION
**Pull request by dpeng123**
PLEASE FILL IN THE PR DESCRIPTION HERE.
#1217 
## Purpose
Enable TeaCache for HiDream-I1-Full through the generic hook backend and a HiDream-specific extractor, using the official TeaCache4HiDream-I1 cache signal (t_embedder output) and polynomial coefficients. Adds recipe/docs updates and extractor unit tests.

### Unit test
```
(main) root@C.45935028:/workspace/vllm-omni$ python -m pytest -q tests/diffusion/cache/test_teacache.py tests/diffusion/cache/test_teacache_extractors.py
.............................                                            [100%]
=============================== warnings summary ===============================
../../usr/local/lib/python3.12/dist-packages/torch/jit/_script.py:365: 14 warnings
  /usr/local/lib/python3.12/dist-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--- Running Summary
29 passed, 14 warnings in 197.45s (0:03:17)
```
### Online smoke test
```
curl -sS -w "\nHTTP_STATUS:%{http_code}\n" http://localhost:8092/v1/images/generations \
  -H "Content-Type: application/json" \
  -d '{
    "model": "HiDream-ai/HiDream-I1-Full",
    "prompt": "The setting sun of late autumn dyes the riverside with a warm orange hue",
    "negative_prompt": "low quality, blurry",
    "n": 1,
    "num_inference_steps": 50,
    "seed": 42,
    "height": 1024,
    "width": 1024,
    "guidance_scale": 5.0
  }'
```
Server evidence:
```
07-26 22:34:14 [backend.py:166] TeaCache applied with rel_l1_thresh=0.2, transformer_class=HiDreamImageTransformer2DModel
INFO: 127.0.0.1:34820 - "POST /v1/images/generations HTTP/1.1" 200 OK
```
result:
```
status: 200
image_size: 1024x1024
image_mode: RGB
output_file: /tmp/hidream_pr5408_online.png
output_bytes: 3147861
e2e_stage_0_wall_time_ms: 29042.135
denoise_step_latency_ms: 580.831
stage_gen_time_ms: 29041.567
```

### Offline A/B test
- model: HiDream-ai/HiDream-I1-Full
- auxiliary_text_encoder: meta-llama/Llama-3.1-8B-Instruct
- prompt: The setting sun of late autumn dyes the riverside with a warm orange hue
- negative_prompt: low quality, blurry
- seed: 42
- size: 1024x1024
- steps: 50
- guidance_scale: 5.0
- GPU: 1x A100-SXM4-80GB
- TeaCache config: rel_l1_thresh=0.2, calibrated HiDream coefficients

| Mode | Exit | Image | Generation time | Denoise step latency | Peak memory metric | nvidia-smi peak | LPIPS vs baseline |
|---|---:|---|---:|---:|---:|---:|---:|
| Baseline no cache | 0 | 1024x1024 RGB | 45.1044s | 902.037 ms | 65100 MB | GPU0 66124 MiB | 0.0000 |
| TeaCache | 0 | 1024x1024 RGB | 29.0283s | 580.520 ms | 64682 MB | GPU0 65706 MiB | 0.0096 |

Result:
- Generation speedup vs baseline: 1.55x.
- Generation time reduced: 35.6%.
- Denoise latency speedup: 1.55x.

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/diffusion/cache/teacache/extractors.py [hunk 1]
```diff
@@ -1187,6 +1187,199 @@ def postprocess(h: torch.Tensor) -> Any:
...
+) -> CacheContext:
+    """Extract cache context for HiDreamImageTransformer2DModel."""
+    from diffusers.models.modeling_outputs import Transformer2DModelOutput
+
+    encoder_hidden_states = kwargs.get("encoder_hidden_states", None)
+    if encoder_hidden_states is not None:
+        encoder_hidden_states_t5 = encoder_hidden_states[0]
+        encoder_hidden_states_llama3 = encoder_hidden_states[1]
+
+    if hidden_states_masks is not None and (img_ids is None or img_sizes is None):
```
Comment (david6666666) [inline vllm_omni/diffusion/cache/teacache/extractors.py:1331]: Can we keep the matching 3D `hidden_states` check from `forward()` here? Enabling TeaCache should not change this input contract.

File: recipes/HiDream/HiDream-I1-Image.md [hunk 2]
```diff
@@ -104,3 +104,24 @@ curl -s http://localhost:8092/v1/chat/completions   -H "Content-Type: applicatio
...
+```bash
+python examples/offline_inference/text_to_image/text_to_image.py \
+  --model HiDream-ai/HiDream-I1-Full \
+  --auxiliary-text-encoder meta-llama/Llama-3.1-8B-Instruct \
+  --prompt "The setting sun of late autumn dyes the riverside with a warm orange hue" \
+  --seed 42 \
+  --height 1024 --width 1024 \
+  --num-inference-steps 50 --guidance-scale 5.0 \
+  --tensor-parallel-size 1 \
+  --vae_use_slicing --vae_use_tiling \
```
Comment (david6666666) [inline recipes/HiDream/HiDream-I1-Image.md]: These options are hyphenated in the parser: `--vae-use-slicing --vae-use-tiling`.

File: vllm_omni/diffusion/cache/teacache/extractors.py (hunk 1 above)
Comment (dpeng123) [reply vllm_omni/diffusion/cache/teacache/extractors.py:1331]: Thanks for reviewing the PR! Updated to add the 3D hidden_states check

File: recipes/HiDream/HiDream-I1-Image.md (hunk 2 above)
Comment (dpeng123) [reply recipes/HiDream/HiDream-I1-Image.md]: Good catch, updated to --vae-use-slicing --vae-use-tiling

File: vllm_omni/diffusion/cache/teacache/extractors.py [hunk 3]
```diff
@@ -1187,6 +1187,204 @@ def postprocess(h: torch.Tensor) -> Any:
...
+    hidden_states: torch.Tensor,
+    timesteps: torch.Tensor,
+    encoder_hidden_states_t5: torch.Tensor,
+    encoder_hidden_states_llama3: torch.Tensor,
+    pooled_embeds: torch.Tensor,
+    img_ids: torch.Tensor | None,
+    img_sizes: list[tuple[int, int]] | None,
+    hidden_states_masks: torch.Tensor | None,
+) -> dict[str, Any]:
+    """Mirror HiDreamImageTransformer2DModel preprocessing for TeaCache."""
```
Comment (lishunyang12) [inline vllm_omni/diffusion/cache/teacache/extractors.py:1201]: Share HiDream preprocessing and block execution with forward() instead of maintaining two copies.
